sidebars.js

```diff
@@ -11,21 +11,37 @@
 
 // @ts-check
 
-/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
-const sidebars = {
-  // By default, Docusaurus generates a sidebar from the docs folder structure
-  tutorialSidebar: [{ type: 'autogenerated', dirName: '.' }],
-
-  // But you can create a sidebar manually
-  /*
-  tutorialSidebar: [
-    {
-      type: 'category',
-      label: 'Tutorial',
-      items: ['hello'],
-    },
-  ],
-   */
+const swap = (items, itemsWithHref) => {
+  const result = items.map((item) => {
+    if (item.type === 'category') {
+      return { ...item, items: swap(item.items, itemsWithHref) }
+    }
+
+    if (itemsWithHref[item.id]) {
+      item = {
+        type: 'link',
+        label: itemsWithHref[item.id].title,
+        href: itemsWithHref[item.id].href,
+      }
+    }
+
+    return item
+  })
+
+  return result
+}
+
+// Switch out doc items for external links where required
+const swapDocItemsToLinkItems = (items, docs) => {
+  const itemsWithHref = {}
+
+  for (const docItem of docs) {
```

Comment:
I have a feeling that if we use `sidebar_custom_props`, this will not be necessary. From [the docs](https://docusaurus.io/docs/sidebar/items#sidebar-item-doc):
> You can use `sidebar_custom_props` to declare custom metadata for a doc page.

> Sidebar custom props is a useful way to propagate arbitrary doc metadata to the client side, so you can get additional information when using any doc-related hook that fetches a doc object.

> `customProps?: Record<string, unknown>`

Reply:
So this didn't work. `customProps` are sent to the component but just become part of the `frontMatter` in this function, so no different to `href`. I'd have to access it with `docItem.frontMatter.sidebar_custom_props.href` which seems a bit pointless when I'm just trying to build `link` type.

They don't magically appear on the generated doc item, as they don't actually get mapped onto doc items: https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-plugin-content-docs/src/sidebars/generator.ts#L139-L152

What this did make me realise though, is that my custom item generator didn't consider all the properties that could be passed to the component.

The `SidebarItemLink` type and `SidebarItemBase` describe these
https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-plugin-content-docs/src/sidebars/types.ts#L43-L48

So I've made sure to map all the properties (except autoAddBaseUrl which seems to be some internal flag because it throws a validation exception if I try to map it).